src/metrics/loc.rs

```diff
@@ -880,6 +889,20 @@ mod tests {
         );
     }
 
+    #[test]
+    fn cpp_lloc() {
+        check_metrics!(
+            "nsTArray<xpcGCCallback> callbacks(extraGCCallbacks.Clone());
+             for (uint32_t i = 0; i < callbacks.Length(); ++i) {
+                 callbacks[i](status);
+             }",
+            "foo.cpp",
+            CppParser,
+            loc,
+            [(lloc, 3, usize)] // nsTArray, for, callbacks
+        );
+    }
```

Comment:
Could you run the minimal tests comparison? Then after checking the changes are good, add a few more tests here.